src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -68,7 +68,7 @@ static TracerProviderSdk()
                 // Callback when Activity is started.
                 ActivityStarted = (activity) =>
                 {
-                    if (activity.IsAllDataRequested)
+                    if (!Sdk.SuppressInstrumentation && activity.IsAllDataRequested)

```

Comment:
we shouldn't need this as we already prevented the activity from being created in the 1st place right?

Reply:
See the info I left about this above ⬆️.

Reply:
Opened https://github.com/dotnet/runtime/issues/41306 to validate whether this is expected or not.

Reply:
Great find @alanwest. I seem to remember seeing some discussion about that behavior somewhere and that is the correct behavior but let's see what they say.